File: helparty/src/main/java/com/hamryt/helparty/dto/gym/GymDTO.java
package com.hamryt.helparty.dto.gym;

import com.hamryt.helparty.dto.UserType;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
public class GymDTO {
    
    private Long id;
    
    private String email;
    
    private String gymName;
    
    private String password;
    
    private String phoneNumber;
    
    private String addressCode;
    
    private String addressDetail;
    
    private UserType userType;
    
    @Builder
    public GymDTO(
        String email, String gymName, String password, String phoneNumber,
        String addressCode, String addressDetail, UserType userType
    ) {
        this.email = email;
        this.password = password;
        this.gymName = gymName;
        this.phoneNumber = phoneNumber;
        this.addressCode = addressCode;
        this.addressDetail = addressDetail;
        this.userType = userType;
    }
    
}

File: helparty/src/main/java/com/hamryt/helparty/dto/gym/GymDTO.kt
package com.hamryt.helparty.dto.gym

import com.hamryt.helparty.dto.UserType

data class GymDTO(
    val id: Long?,
    val email: String,
    val gymName: String,
    val password: String,
    val phoneNumber: String,
    val addressCode: String,
    val addressDetail: String,
    val userType: UserType
) {
    companion object {
        @JvmStatic
        fun create(
            id: Long?, email: String, gymName: String,
            password: String, phoneNumber: String, addressCode: String,
            addressDetail: String, userType: UserType
        ) =
            GymDTO(id, email, gymName, password, phoneNumber, addressCode, addressDetail, userType)

    }
}
